tests/testlib/s2n_extension_test_utils.c

```diff
@@ -24,3 +25,18 @@ int s2n_connection_allow_all_response_extensions(struct s2n_connection *conn)
     POSIX_CHECKED_MEMSET(&conn->extension_requests_sent, 0xFF, S2N_SUPPORTED_EXTENSIONS_BITFIELD_LEN);
     return S2N_SUCCESS;
 }
+
+int s2n_connection_allow_response_extension(struct s2n_connection *conn, uint16_t iana_value)
+{
+    s2n_extension_type_id extension_id = s2n_unsupported_extension;
+    POSIX_GUARD(s2n_extension_supported_iana_value_to_id(iana_value, &extension_id));
+    S2N_CBIT_SET(conn->extension_responses_received, extension_id);
+}
+
+int s2n_connection_allow_request_extension(struct s2n_connection *conn, uint16_t iana_value)
```

Comment:
I don't think these names are right. Request extensions are always allowed, and s2n_connection_allow_response_extension isn't actually allowing responses, it's marking that they already happened.

You definitely need to change s2n_connection_allow_request_extension to s2n_connection_allow_response_extension. For s2n_connection_allow_response_extension, I think a more accurate name might be s2n_connection_mock_response_extension or s2n_connection_mark_response_extension_received.

Reply:
Ah okay, I think I understand. I changed the names in 183079f